server/src/main/java/org/elasticsearch/cluster/InternalClusterInfoService.java

```diff
@@ -183,147 +373,49 @@ public ClusterInfo getClusterInfo() {
             indicesStatsSummary.shardSizes, indicesStatsSummary.shardRoutingToDataPath, indicesStatsSummary.reservedSpace);
     }
 
-    /**
-     * Retrieve the latest nodes stats, calling the listener when complete
-     * @return a latch that can be used to wait for the nodes stats to complete if desired
-     */
-    protected CountDownLatch updateNodeStats(final ActionListener<NodesStatsResponse> listener) {
-        final CountDownLatch latch = new CountDownLatch(1);
-        final NodesStatsRequest nodesStatsRequest = new NodesStatsRequest("data:true");
-        nodesStatsRequest.clear();
-        nodesStatsRequest.addMetric(NodesStatsRequest.Metric.FS.metricName());
-        nodesStatsRequest.timeout(fetchTimeout);
-        client.admin().cluster().nodesStats(nodesStatsRequest, new LatchedActionListener<>(listener, latch));
-        return latch;
-    }
-
-    /**
-     * Retrieve the latest indices stats, calling the listener when complete
-     * @return a latch that can be used to wait for the indices stats to complete if desired
-     */
-    protected CountDownLatch updateIndicesStats(final ActionListener<IndicesStatsResponse> listener) {
-        final CountDownLatch latch = new CountDownLatch(1);
-        final IndicesStatsRequest indicesStatsRequest = new IndicesStatsRequest();
-        indicesStatsRequest.clear();
-        indicesStatsRequest.store(true);
-        indicesStatsRequest.indicesOptions(IndicesOptions.STRICT_EXPAND_OPEN_CLOSED_HIDDEN);
-
-        client.admin().indices().stats(indicesStatsRequest, new LatchedActionListener<>(listener, latch));
-        return latch;
-    }
-
     // allow tests to adjust the node stats on receipt
     List<NodeStats> adjustNodesStats(List<NodeStats> nodeStats) {
         return nodeStats;
     }
 
     /**
-     * Refreshes the ClusterInfo in a blocking fashion
+     * Refreshes the ClusterInfo in a blocking fashion, for use in tests
      */
-    public final ClusterInfo refresh() {
+    public ClusterInfo refresh() {
```

Comment:
can we remove this method from the class and move it to a test class? (e.g. in `test:framework`)

Reply:
Yep, see 9b42365.